osu.Server.Spectator/StartupDevelopment.cs

```diff
@@ -45,7 +45,8 @@ public class LocalAuthenticationHandler : AuthenticationHandler<AuthenticationSc
         /// </summary>
         protected override Task<AuthenticateResult> HandleAuthenticateAsync()
         {
-            if (!Context.Request.Headers.TryGetValue("user_id", out var userIdString))
+            string? userIdString;
+            if (!Context.Request.Headers.TryGetValue("user_id", out var userIdValues) || (userIdString = userIdValues) == null)
```

Comment:
This compound-assignment-and-nullcheck is dreadful readability-wise. If you ask me this should read

```csharp
string? userIdString = null;

if (Context.Request.Headers.TryGetValue("user_id", out var userIdValue))
    userIdString = userIdValue;

userIdString ??= Interlocked.Increment(ref userIDCounter).ToString();
```

Reply:
I agree, way better.